File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        result = 0
        charSet = set(s)

        for c in charSet:
            count = 0
            left = 0

            for right in range(len(s)):
                if s[right] == c:
                    count += 1
                
                while (right - left + 1) - count > k:
                    if s[left] == c:
                        count -= 1
                    left += 1
                result = max(result, right - left + 1)
        return result

File: Data Structures & Algorithms/jump-game-ii/submission-0.java
class Solution {
    public int jump(int[] nums) {
        int l = 0;
        int r = 0;
        int result = 0;

        while (r < nums.length - 1) {
            int distance = 0;

            for(int i = l; i < r + 1; i++) {
                distance = Math.max(distance, i + nums[i]);
            }

            l = r + 1;
            r = distance;
            result++;
        }

        return result;
    }
}
